fhttp/http_client.go

```diff
@@ -176,14 +176,15 @@ type HTTPOptions struct {
 	// ExtraHeaders to be added to each request (UserAgent and headers set through AddAndValidateExtraHeader()).
 	extraHeaders http.Header
 	// Host is treated specially, remember that virtual header separately.
-	hostOverride     string
-	HTTPReqTimeOut   time.Duration // timeout value for http request
-	UserCredentials  string        // user credentials for authorization
-	ContentType      string        // indicates request body type, implies POST instead of GET
-	Payload          []byte        // body for http request, implies POST if not empty.
-	LogErrors        bool          // whether to log non 2xx code as they occur or not
-	ID               int           // id to use for logging (thread id when used as a runner)
-	SequentialWarmup bool          // whether to do http(s) runs warmup sequentially or in parallel (new default is //)
+	hostOverride       string
+	HTTPReqTimeOut     time.Duration // timeout value for http request
+	UserCredentials    string        // user credentials for authorization
+	ContentType        string        // indicates request body type, implies POST instead of GET
+	Payload            []byte        // body for http request, implies POST if not empty.
+	LogErrors          bool          // whether to log non 2xx code as they occur or not
+	ID                 int           // id to use for logging (thread id when used as a runner)
+	SequentialWarmup   bool          // whether to do http(s) runs warmup sequentially or in parallel (new default is //)
+	MaxConnectionReuse []int         // range of max number of connection to reuse for each thread.
```

Comment:
[2]int if it's always a range? or an actual pair struct? ([2]int seems simple enough)
though I guess you use empty range for infinite? [comment as much either way even if leaving as a slice:
empty slice = no limit; otherwise must be a slice of 2..?)

Reply:
updated